msbuild/Xamarin.MacDev.Tasks/Tasks/XcodeBuildTask.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Collections.Generic;
+
+namespace Xamarin.MacDev.Tasks {
+	public abstract class XcodeBuildTask : XamarinTask {
+
+		// Task input parameters
+		public string SdkDevPath { get; set; } = string.Empty;
+
+		public string WorkingDirectory { get; set; } = string.Empty;
+
+		public string OutputPath { get; set; } = string.Empty;
+
+
+		protected virtual string Command { get; set; } = string.Empty;
+
+		protected virtual IList<string> GenerateCommandLineCommands () => new List<string> ();
```

Comment:
Array does implement IList in c# the following will do the same without the allocation:
```csharp
protected virtual IList<string> GenerateCommandLineCommands () => Array.Empty<string> ();
```